gulpfile.js

```diff
@@ -0,0 +1,99 @@
+/* eslint-env node */
+
+const gulp = require("gulp");
+const sass = require("gulp-sass")(require("sass"));
+const rename = require("gulp-rename");
+const gulpRev = require("gulp-rev-all");
+const collectFiles = require("./scripts/gulp/collect-files");
+const replace = require("gulp-replace");
+
+const SASS_FILES = {
+  "sass/hds/index.scss": { basename: "hds" },
+  "sass/shared/index.scss": { basename: "shared" },
+  "sass/login/index.scss": { basename: "hs-login" },
+};
+const STYLES_THEME_PROPERTIES_KEY = "styles";
+
+const SCRIPTS_THEME_PROPERTIES_KEY = "scripts";
+
+function trimFromStart(str, prefix) {
+  return str.startsWith(prefix) ? str.substring(prefix.length) : str;
+}
+
+function styles() {
+  const sourceFiles = Object.keys(SASS_FILES);
+
+  return gulp
+    .src(sourceFiles)
+    .pipe(sass.sync().on("error", sass.logError))
+    .pipe(
+      gulpRev.revision({
+        transformFilename: (file /*, hash*/) => {
+          return file.basename;
+        },
+      })
+    )
+    .pipe(
+      rename((path, file) => {
+        const originalFullPath = file.history[0];
+        const originalFileName = trimFromStart(
+          originalFullPath,
+          `${file.cwd}/`
+        );
+
+        path.dirname = "";
+        path.basename = SASS_FILES[originalFileName].basename;
+      })
+    )
+    .pipe(collectFiles(STYLES_THEME_PROPERTIES_KEY))
+    .pipe(gulp.dest("./helsinki/login/resources/css"));
+}
+
+function scripts() {
+  return gulp
+    .src("helsinki/login/resources/js/*")
+    .pipe(
+      gulpRev.revision({
+        transformFilename: (file /*, hash*/) => {
+          return file.basename;
+        },
+      })
+    )
+    .pipe(collectFiles(SCRIPTS_THEME_PROPERTIES_KEY));
+}
```

Comment:
I'm trying to understand why files are collected to grouped. Maybe because results of `scripts()` and `styles()` are both passed to `themePropertiesStyles` and `themePropertiesScripts`?

Files are already collected in `src()`, so why store them to a variable in `collect-files.js` and later on retrieve them?

Could the whole collection process be removed, if the `exports.default` is changed to process styles and scripts separately?